nmpc_params_node:
  ros__parameters:
    lowX: [-6.0, -6.0, -3.14]
    highX: [6.0,  6.0,  3.14]
    lowU: -20.0
    highU: 20.0
    mat_Q: [1000, 1000, 2000]
    mat_R: [1, 1, 1, 1]
    prediction_horizons: 50
    step_time: 0.1
    mpciter: 0
    mpc_timer: 0.1
    control_timer: 0.01
    wheel_topic: "/odom_wheel"
    imu_topic: "/imu/data2"
    control_topic: "/input_controls"
    path_topic: "/beizer_path"
    current_state: [0.0, 0.0, 0.0]
    current_control: [0.0, 0.0, 0.0, 0.0]
    goal_path: [0.0, 0.0, 0.0]
    offset: 3.0
    search_index: 10
    previous_index: 0
    next_index: 0
    index: 0
    dind: 0
    startX: [0.0, 0.0, 0.0]
    endX: [0.0, 0.0, 0.0]
    goal_flag: False
    hold_state: True
    opt_control: [0.0, 0.0, 0.0, 0.0]
    norm_cond: 0.0
    n_points: 100
